### ISSUE TITLE
`comma-dangle`: should not insert comma in arrow function return expressions

### ISSUE THREAD

**Issue by noelforte** [eslint-stylistic/eslint-stylistic#300]
### Validations

- [X] Follow our [Code of Conduct](https://github.com/eslint-stylistic/eslint-stylistic/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guide](https://eslint.style/contribute/guide).
- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.

### Describe the bug

Possibly related: #158 / #160 

When writing arrow functions with an implicit return, sometimes it's useful to drop onto the next line to get more space to write the expression. Prettier uses this pattern a lot:

```js
const nums = [1, 2, 3];

const remappedArray = nums.map(
  (el) => `${el} plus 5 is ${el + 5}`
);
```

However when using a callback function that also happens to be an arrow function with an implicit return statement, such as when working with arrays, patterns like:

```js
(el) => `${el} plus 5 is ${el + 5}`
```

get formatted to:
```js
(el) => `${el} plus 5 is ${el + 5}`,
```

While valid, it is somewhat confusing because a line ending with a comma is not allowed in a return statement, but is allowed in an expression via the comma operator...**but** neither of those cases are happening here becuase `Array.map` takes 2 arguments, the callback and [`thisArg`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map#thisarg). So, in this case JavaScript is interpreting the above as:

```js
... nums.map( 
  ((el) => `${el} plus 5 is ${el + 5}`),
) ...
```

Since the comma operator doesn't seem to be as well-known of a feature and the correct way to use it in an arrow function is actually `() => (exp1, exp2, ...)`, perhaps the best way to handle this would be to not insert dangling commas in places where their meaning is ambiguous.

Happy to submit a PR with failing tests, however Vitest isn't something I'm as familiar with so I don't know how much help I could be. Still willing to give it a shot though!

### Reproduction

https://stackblitz.com/edit/github-2cdzyb?file=src%2Findex.ts&view=editor

### Contributes

- [ ] I am willing to submit a PR to fix this issue
- [X] I am willing to submit a PR with failing tests

**Comment by specious** [comment]
I chose to relax the `comma-dangle` rule from `always-multiline` to `only-multiline` just to avoid the warning generated by this situation for the time being, since there doesn't appear to be a way to exclude just this case.

```js
const lines = content.split('\n').filter(line =>
  line.trim() && !keysToRemove.some(key => line.startsWith(`${key}=`))
)
```

Was generating this warning:

```
/path/to/file.js
  14:71  warning  Missing trailing comma  @stylistic/js/comma-dangle
```

This means that the linter will now not warn when trailing commas are not used in multi-line definitions.
